Source/WebKit/UIProcess/Cocoa/FullscreenClient.mm

```diff
@@ -72,6 +76,14 @@
 
 void FullscreenClient::didEnterFullscreen(WebPageProxy*)
 {
+#if ENABLE(QUICKLOOK_FULLSCREEN)
+    auto fullScreenController = [m_webView fullScreenWindowController];
+    if (fullScreenController) {
+        CGSize imageDimensions = fullScreenController.imageDimensions;
+        if (fullScreenController.isUsingQuickLook && m_delegateMethods.webViewDidFullscreenImageWithQuickLook)
+            [m_delegate.get() _webView:m_webView didFullscreenImageWithQuickLook:imageDimensions];
+    }
+#endif // ENABLE(QUICKLOOK_FULLSCREEN)
```

Comment:
I would move this added logic to the end of the method. At the very least, I think it should not be called before `[m_webView willChangeValueForKey:@"fullscreenState"]`.